config/src/config/network_config.rs

```diff
@@ -240,10 +261,13 @@ impl NetworkConfig {
         self.identity = Identity::from_config(identity_key, peer_id);
     }
 
-    /// Check that all seed peer addresses look like canonical DiemNet addresses
-    pub fn verify_seed_addrs(&self) -> Result<(), Error> {
-        for (peer_id, addrs) in self.seed_addrs.iter() {
-            for addr in addrs {
+    pub fn verify_seeds(&self) -> Result<(), Error> {
+        for (peer_id, seed) in self.seeds.iter() {
+            crate::config::invariant(
+                !seed.keys.is_empty(),
+                format!("Seed peer {} has no pubkeys", peer_id.short_str()),
+            )?;
```

Comment:
My mistake, probably moved too quickly to notice I did this

Reply:
So, I've changed the invariant now.  It must either have an Address with a key, or pubkeys specified.  Since later, address pubkeys are merged with pubkeys.